src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs

```diff
@@ -97,99 +97,104 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
 
             var message = formatter(state, exception);
 
-            if (!string.IsNullOrEmpty(message))
-            {
-                WriteMessage(logLevel, Name, eventId.Id, message);
-            }
-
-            if (exception != null)
+            if (!string.IsNullOrEmpty(message) || exception != null)
             {
-                WriteException(logLevel, Name, eventId.Id, exception);
+                WriteMessage(logLevel, Name, eventId.Id, message, exception);
             }
         }
 
-        public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId, string message)
+        public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId, string message, Exception exception)
         {
-            // check if the message has any new line characters in it and provide the padding if necessary
-            message = ReplaceMessageNewLinesWithPadding(message);
-            var logLevelColors = GetLogLevelConsoleColors(logLevel);
-            var loglevelString = GetLogLevelString(logLevel);
+            var logLevelColors = default(ConsoleColors);
+            var logLevelString = string.Empty;
+            var logIdentifier = string.Empty;
+            var scopeInformation = string.Empty;
+            var exceptionText = string.Empty;
+            var printMessage = false;
+            var printException = false;
 
             // Example:
             // INFO: ConsoleApp.Program[10]
             //       Request received
-
-            lock (_lock)
+            if (!string.IsNullOrEmpty(message))
             {
-                // log level string
-                WriteWithColor(
-                    logLevelColors.Foreground,
-                    logLevelColors.Background,
-                    loglevelString,
-                    newLine: false);
-
+                logLevelColors = GetLogLevelConsoleColors(logLevel);
+                logLevelString = GetLogLevelString(logLevel);
                 // category and event id
-                // use default colors
-                WriteWithColor(
-                    DefaultConsoleColor,
-                    DefaultConsoleColor,
-                    _loglevelPadding + logName + $"[{eventId}]",
-                    newLine: true);
-
+                logIdentifier = _loglevelPadding + logName + "[" + eventId + "]";
                 // scope information
                 if (IncludeScopes)
                 {
-                    var scopeInformation = GetScopeInformation();
+                    scopeInformation = GetScopeInformation();
```

Comment:
`scopeInformation` is going to access TLS/ExecutionContext. It's the opposite of the usual problem of threading. 

> another thread could modify the scope tree information giving 

This won't happen. The scope information is not accessible to another thread.